src/Data.cpp

```diff
@@ -0,0 +1,494 @@
+      flexbounds_.at("d_l").second - flexbounds_.at("d_l").first + 1);
+  Eigen::MatrixXd nmatch_matrix =
+      GermlineTransProbMatrix(dgerm_ptr, "d_l", "d_r");
+
+  // Multiply in the associated landing-out and gene probabilities.
+  MultiplyLandingMatchMatrix(dgerm_ptr->landing_out(), dgerm_ptr->name(), "v_r",
+                             false, xmatch_matrix);
+  MultiplyLandingMatchMatrix(dgerm_ptr->landing_out(), dgerm_ptr->name(), "d_l",
+                             false, nmatch_matrix);
+  xmatch_matrix *= dgerm_ptr->gene_prob();
+  nmatch_matrix *= dgerm_ptr->gene_prob();
+
+  // Construct germline match matrices with per-site emission probabilities.
+  Eigen::MatrixXd emission_xmatch_matrix =
+      EmissionMatchMatrix(dgerm_ptr, "v_r", xmatch_matrix);
+  Eigen::MatrixXd emission_nmatch_matrix =
+      EmissionMatchMatrix(dgerm_ptr, "d_l", nmatch_matrix);
+
+  // Store Smooshable objects.
+  SmooshablePtrVect dx_smooshable = {BuildSmooshablePtr(
+      dgerm_ptr, "v_r", xmatch_matrix, emission_xmatch_matrix)};
+  SmooshablePtrVect dn_smooshables = {
+      BuildSmooshablePtr(nullptr, "", Eigen::MatrixXd::Zero(0, 0), nti_matrix),
+      BuildSmooshablePtr(dgerm_ptr, "d_l", nmatch_matrix,
+                         emission_nmatch_matrix)};
+
+  return std::make_pair(dx_smooshable, dn_smooshables);
+};
+
+
+/// @brief Creates Smooshable objects for a given J germline gene and read/MSA.
+/// @param[in] jgerm_ptr
+/// A pointer to an object of class JGermline.
+/// @return
+/// A 2-tuple containing a SmooshablePtrVect of size 1 (for the non-NTI case)
+/// and a SmooshablePtrVect of size 2 (for the NTI case).
+std::pair<SmooshablePtrVect, SmooshablePtrVect> Data::JSmooshables(
+    JGermlinePtr jgerm_ptr) const {
+  // Compute the transition probability portion of the germline match matrix
+  // (assuming no left-NTIs).
+  Eigen::MatrixXd xmatch_matrix =
+      GermlineTransProbMatrix(jgerm_ptr, "d_r", "j_r");
+
+  // Multiply in the associated landing-in probabilities (if necessary).
+  if (relpos_.at(jgerm_ptr->name()) <= flexbounds_.at("d_r").second) {
+    MultiplyLandingMatchMatrix(jgerm_ptr->landing_in(), jgerm_ptr->name(),
+                               "d_r", true, xmatch_matrix);
+  }
+
+  // Compute the transition probability portion of the germline match matrix
+  // (assuming some left-NTIs).
+  // Eigen::MatrixXd nti_prob_matrix = jgerm_data.NTIProbMatrix(
+  //     flexbounds.at("d_r"), flexbounds.at("j_l"), emission_indices,
+  //     j_relpos);
+  Eigen::MatrixXd nti_matrix = Eigen::MatrixXd::Ones(
+      flexbounds_.at("d_r").second - flexbounds_.at("d_r").first + 1,
+      flexbounds_.at("j_l").second - flexbounds_.at("j_l").first + 1);
+  Eigen::MatrixXd nmatch_matrix =
+      GermlineTransProbMatrix(jgerm_ptr, "j_l", "j_r");
+
+  // Extract the column that corresponds to the last match ending position with
+  // a germline state.
+  // int end_pos = std::min(flexbounds_.at("j_r").second,
```

Comment:
Yes, I hadn't dealt with this start/end stuff yet, but I envisioned giving the Smooshable row and column indices that tell the Smooshable what rows/columns of `marginal_` should be extracted. 

This passing of indices (I think?) is necessary because we're having to continually re-update V and J marginal matrices and we need the full match matrix to do this.  So we'd recompute a full matrix, and cut down again.  We could talk about this sometime.